### PR TITLE
Save state after device rotation

### PR DESCRIPTION
**Pull request by ypdieguez**
Creation of a dialog that remains visible and conserves the state after a device rotation, in case that activity does not have the following attributes in the manifest:
```
android:configChanges = "orientation | keyboardHidden | screenSize"
android:screenOrientation = "portrait"
```

### CODE REVIEW

**Comment by mattsilber** [comment]
First, thank you so much for contributing, and 👍 on beating me to all of the annotations, but there are a couple things I wanted to discuss (which are still totally up for debate, but I have no one else to talk with):

I explicitly never add kotlin/androidx to the library files, as I do not want to force users to require using them. I know it's the de facto language moving forward (and it's my favorite), but there was nothing complex enough in the original package for me to justify that requirement. I've been considering additional `-ktx` or `-androidx` library packages, but the interopability with the core seemed fine in its default sense. Even the minSdk bump from 9 to 14 (even though it's recommended) and lambda support are some things I was devoutly trying to avoid. 

Basically, I'm against killing compatibility support until we can no longer compile against those packages (which androidx would officially do), because I don't believe it should be the library's choice to force people to upgrade when not truly necessary. Would love to hear your thoughts on this.

But I love the drastic cutdown of the bloat too, great work and thanks again for contributing!

File: sigcap/src/main/java/com/guardanis/sigcap/SignatureInputView.java [hunk 1]
```diff
@@ -134,19 +138,41 @@ public SignatureResponse saveSignature() {
...
 
     public SignatureRenderer getSignatureRenderer() {
         return renderer;
     }
+
+    /**
+     * Get signature paths.
+     * @author Yordan P. Dieguez
+     */
+    public List<List<Path>> getSignaturePaths() {
```
Comment (mattsilber) [inline sigcap/src/main/java/com/guardanis/sigcap/SignatureInputView.java:165]: I never exposed these List of Paths because their backing array is mutable. If we want to open this up, we should probably be returning cloned instances of these Lists instead

File: sigcap/src/main/java/com/guardanis/sigcap/SignatureDialogBuilder.java [hunk 2]
```diff
@@ -1,80 +1,57 @@
...
+ * Helper class to show dialog.
+ * To use it, the activity or parent fragment must implement {@link SignatureEventListener} interface,
+ * otherwise it will throw an {@link RuntimeException}
+ *
+ * @author Matt Silber
+ * @author Yordan P. Dieguez
+ */
 public class SignatureDialogBuilder {
 
-    public interface SignatureEventListener {
```
Comment (mattsilber) [inline sigcap/src/main/java/com/guardanis/sigcap/SignatureDialogBuilder.java:10]: 👍 on pulling that out

**Comment by ypdieguez** [comment]
Hello, first thanks for this library, great work. For me, it has been very useful and the best that I have seen on github for this purpose.:+1:

I tell you what happened to me when using it:
In the app I'm developing I'm using the single activity app architecture (Google officially recommends), and my activity does not have the following attributes in the manifest `android:configChanges = "orientation | keyboardHidden | screenSize"` and `android:screenOrientation = "portrait"`, because I need that my layouts refresh if the screen rotates.

So, What happens if the screen rotates?: :worried:
1. `SignatureInputView` loses its state because it is initialized again.

2. If `SignatureDialogBuilder` is used to show the dialog, then it is dismissed (and in the case that still showing, the listener is lost as it is currently implemented).

:point_right: To solve the first problem, I first tried to use the classic `onSaveInstanceState` and `onRestoreInstanceState` methods, but the `Path` class is not `Parcelable`. So the alternative I found was to return the `signaturePaths` to save it (with `ViewModel`) and then pass it again to restore.

:point_right: To solve the second problem, I had to use `DialogFragment` and force the parent fragment or activity to implement the interface `SignatureEventListener`.

As you can see, I use `DialogFragment` and `ViewModel` that generate dependencies, plus when using in `build.gradle` the  `compileSdkVersion 29` setting practically guides me to use `androidx`.

This was the solution I found, there are possibly other ways to do better. What would be good is to solve the issues commented above. 

Points to consider:
- Regarding using kotlin, JDK8 and lambdas it was a personal matter, I think that there are still many android developers that do not use it and you would be absolutely right not to force them to use it.
- On `minSdkVersion 9`, let's be honest, few developers would be concerned about this, they would quickly move their minimum sdk to 14.
- I really think it is time to move forward, the semantic versioning is the one that defines the compatibility between the projects and the library.

**P.S.** If it is not possible to complete the pull request or find other solution more appropriate, I would like if you can at least leave the methods `getSignaturePaths` and `setSignaturePaths` in `SignatureInputView ` and publish a new version, so I can make a wrapper of your library. :relaxed:

File: sigcap/src/main/java/com/guardanis/sigcap/SignatureInputView.java (hunk 1 above)
Comment (ypdieguez) [reply sigcap/src/main/java/com/guardanis/sigcap/SignatureInputView.java:165]: It would be a good option :+1:

File: sigcap/src/main/java/com/guardanis/sigcap/SignatureDialogBuilder.java (hunk 2 above)
Comment (ypdieguez) [reply sigcap/src/main/java/com/guardanis/sigcap/SignatureDialogBuilder.java:10]: Thanks :blush:

**Comment by mattsilber** [comment]
Thanks for the detailed response! I think you make some good points, and you're highlighting a pretty significant issue with the code. I'm going to merge this branch in later and then my plan is to:

- Wrap the Path list representations in a serializable entity while not affecting rendering speed
- Properly implement the View's save/restore components for default storage not solely limited to the dialog implementation 
- Update the exposed get-Path methods to return new Path instances from that newly-serialized data so you always have shadow-access
- Update the exposed set-Path methods to integrate with the serialized system
- Create a new `-androidx` module containing an AppCompat variant of the dialog implementation
- Drop the kotlin/androidx requirement from the core back to what it was (the minSdk would naturally revert back to 9 at that point)

That should cover everything needed, while offering both backwards compatibility and integrated support for newer packages
